src/Routing/Middleware/SecurityMiddleware.php

```diff
@@ -0,0 +1,197 @@
+    public function setReferrerPolicy($policy = 'same-origin')
+    {
+        $available = [
+            'no-referrer', 'no-referrer-when-downgrade', 'origin',
+            'origin-when-cross-origin',
+            'same-origin', 'strict-origin', 'strict-origin-when-cross-origin',
+            'unsafe-url'
+        ];
+
+        $this->checkValues($policy, $available);
+        $this->headers['referrer-policy'] = $policy;
+
+        return $this;
+    }
+
+    /**
+     * X-Frame-Options
+     *
+     * Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+     *
+     * Available Value: 'deny', 'sameorigin', 'allow-from <uri>'
+     *
+     * @param string $option Option value
+     * @param string $url URL if mode is `allow-from`
+     * @return $this
+     */
+    public function setXFrameOptions($option = 'sameorigin', $url = null)
+    {
+        $this->checkValues($option, ['deny', 'sameorigin', 'allow-from']);
+
+        if ($option === 'allow-from') {
+            if (empty($url)) {
+                throw new InvalidArgumentException('The 2nd arg $url can not be empty when `allow-from` is used');
+            }
+            $option .= ' ' . $url;
+        }
+
+        $this->headers['x-frame-options'] = $option;
+
+        return $this;
+    }
+
+    /**
+     * X-XSS-Protection
+     *
+     * Reference: https://blogs.msdn.microsoft.com/ieinternals/2011/01/31/controlling-the-xss-filter
+     *
+     * Available Value: '1', '0', '1; mode=block'
+     *
+     * @param string $mode Mode value
+     * @return $this
+     */
+    public function setXssProtection($mode = 'block')
+    {
+        $mode = (string)$mode;
+
+        if ($mode === 'block') {
+            $mode = '1; mode=block';
+        }
+
+        $this->checkValues($mode, ['1', '0', '1; mode=block']);
+        $this->headers['x-permitted-cross-domain-policies'] = $mode;
+
+        return $this;
+    }
+
+    /**
+     * X-Permitted-Cross-Domain-Policies
+     *
+     * Reference: https://www.adobe.com/devnet/adobe-media-server/articles/cross-domain-xml-for-streaming.html
+     *
+     * Available Value: 'all', 'none', 'master-only', 'by-content-type', 'by-ftp-filename'
+     *
+     * @param string $policy Policy value
+     * @return $this
+     */
+    public function setCrossDomainPolicy($policy = 'all')
+    {
+        $this->checkValues($policy, ['all', 'none', 'master-only', 'by-content-type', 'by-ftp-filename']);
+        $this->headers['x-permitted-cross-domain-policies'] = $policy;
+
+        return $this;
+    }
+
+    /**
+     * Convenience method to check if a value is in the list of allowed args
+     *
+     * @throws \InvalidArgumentException Thrown when a value is invalid.
+     * @param string $value
+     * @param array $allowed
+     * @return void
+     */
+    protected function checkValues($value, array $allowed)
+    {
+        if (!in_array($value, $allowed)) {
+            throw new InvalidArgumentException(sprintf(
```

Comment:
Nor do I have here. Do really have to put the whole code into one line to make it less readable but to pass stickler?